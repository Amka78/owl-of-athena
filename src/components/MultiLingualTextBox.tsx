import React, { FunctionComponent } from "react";
import { TextInputProps } from "react-native";
import { TextInput } from "react-native-paper";
import { Dimens, Fonts, Message, Layout, Colors } from "../constants";

type MultiLingualTextBoxProps = TextInputProps & { label?: string };

const textInputStyle = {
    backgroundColor: "transparent",
    marginBottom: Dimens.button_margin_bottom,
    fontFamily: Fonts.primaryRegular,
    fontSize: Dimens.input_text_size,
    width: Layout.window.width - Dimens.content_margin_horizontal * 2
};
export const MultiLingualTextBox: FunctionComponent<
    MultiLingualTextBoxProps
> = (props: MultiLingualTextBoxProps) => {
    return (
        <TextInput
            {...props}
            mode={"flat"}
            style={textInputStyle}
            underlineColor={Colors.white}
            placeholder={
                props.placeholder ? Message.get(props.placeholder) : undefined
            }
            label={props.label ? Message.get(props.label) : undefined}
            theme={{
                colors: {
                    text: Colors.white,
                    placeholder: Colors.white,
                    primary: Colors.cyan
                }
            }}
        ></TextInput>
    );
};
