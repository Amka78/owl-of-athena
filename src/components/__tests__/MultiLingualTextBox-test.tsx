import "react-native";

import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { MultiLingualTextBox } from "../MultiLingualTextBox";

describe("TextBoxEx UnitTest", () => {
    beforeAll(async () => {
        jest.mock("Animated", () => ({
            Value: jest.fn(() => ({
                interpolate: jest.fn(),
                setValue: jest.fn()
            })),
            timing: jest.fn(() => ({
                start: jest.fn()
            }))
        }));
    });

    it("renders correctly", () => {
        const tree = toJson(shallow(<MultiLingualTextBox />));

        expect(tree).toMatchSnapshot();
    });

    afterAll(async () => {
        jest.dontMock("Animated");
    });
});
