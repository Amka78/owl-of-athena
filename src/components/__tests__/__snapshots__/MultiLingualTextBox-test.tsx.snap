// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBoxEx UnitTest renders correctly 1`] = `
<ForwardRef(Themed.Input)
  autoCapitalize="none"
  autoCorrect={false}
  errorStyle={
    Object {
      "fontSize": 12,
      "textAlign": "center",
    }
  }
  keyboardAppearance="light"
  labelStyle={
    Object {
      "alignContent": "flex-start",
      "color": "black",
      "fontFamily": "mont-regular",
    }
  }
  onEndEditing={[Function]}
/>
`;
